lib/ddtrace/contrib/active_support/notifications/subscription.rb

```diff
@@ -28,18 +28,17 @@ def after_trace(&block)
             @after_trace_callbacks << block if block_given?
           end
 
+          def call(name, start, _, id, payload)
+            start_span(start)
+            finish_span(name, id, payload)
```

Comment:
I think there's a bug here: `finish_span` would be applying the wrong finish time.

My suggestion here would be to:

```ruby
def call(name, start, finish, id, payload)
  start_span(name, id, payload, start)
  finish_span(name, id, payload, finish)
end

def start(name, id, payload)
  start_span(name, id, payload)
end

def finish(name, id, payload)
  finish_span(name, id, payload)
end

protected

def start_span(name, id, payload, start = nil)
  # Run callbacks
  run_callbacks(@before_trace_callbacks)

  # Start a trace
  tracer.trace(@span_name, @options).tap do |span|
    # Assign start time if provided
    span.start_time = start unless start.nil?
  end
end

def end_span(name, id, payload, finish = nil)
  tracer.active_span.tap do |span|
    # If no active span, return.
    return nil if span.nil?

    # Call block for event
    block.call(span, name, id, payload)

    # Assign finish time if provided. Else finish.
    finish.nil? ? span.finish : span.finish(finish)

    # Run callbacks
    run_callbacks(@after_trace_callbacks)
  end
end
```

Reply:
Thinking on the example above, we might want to add some error handling around `block.call` too, so an error in that block doesn't break the trace.

Reply:
@skisulli I don't think we should raise or set any kind of error, since generating trace data is meta, and shouldn't break legitimate user code if it fails. When tracing fails, we instead use the tracer logger to quietly log errors.

I'm going to open (another) PR to implement this change (with the `call`) too, because I think this is important for ASN subscription, and I'd like to isolate it from ActiveModelSerializers. We'll rebase on top of that once its merged.